src/DynamoApplications/StartupUtils.cs

```diff
@@ -174,6 +179,64 @@ private static IUpdateManager InitializeUpdateManager()
             return um;
         }
 
+        /// <summary>
+        /// Extracts version of ASM dlls from a path by scanning for ASM dlls in the path.
+        /// Throws if ASM binaries cannot be found in the path.
+        /// </summary>
+        /// <param name="asmPath">path to directory containing asm dlls</param>
+        /// <returns></returns>
+        /// <param name="searchPattern">optional - to be used for testing</param>
+        /// <returns></returns>
+        internal static Version GetVersionFromASMPath(string asmPath, string searchPattern = "ASMAHL*.dll")
+        {
+            var ASMFilePath = Directory.GetFiles(asmPath, searchPattern, SearchOption.TopDirectoryOnly).FirstOrDefault();
+            if(ASMFilePath != null && File.Exists(ASMFilePath))
+            {
+                var asmVersion = FileVersionInfo.GetVersionInfo(ASMFilePath);
+                var libGversion = new Version(asmVersion.FileMajorPart, asmVersion.FileMinorPart, asmVersion.FileBuildPart);
+                return libGversion;
+            }
+            throw new FileNotFoundException("$Could not find geometry library binaries at : {geometryLibraryPath}");
```

Comment:
Remind me, seems at this stage, we do not have a DynamoModel yet, nor logger. Is there a better way to catch this for user and us?